src/Response/ResponseFactory.php

```diff
@@ -0,0 +1,37 @@
+<?php
+declare(strict_types=1);
+
+namespace Mezzio\Response;
+
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * @internal
+ * @deprecated Will be removed with v4.0.0
+ */
+final class ResponseFactory implements ResponseFactoryInterface
```

Comment:
I'd argue for the name `CallableResponseFactoryDecorator` here. Seeing the name, my first inclination seeing the name was "oh my, we're duplicating what we do in Diactoros here?" ... but that's not the case. It's a decorator for a callable that can create a response.